src/main/scala/beam/sim/BeamConfigChangesObservable.scala

```diff
@@ -72,14 +73,12 @@ class BeamConfigChangesObservable @Inject() (beamConfig: BeamConfig) {
     }
   }
 
-  val observable = new WeaklyObservable()
+  private val observable = new WeaklyObservable()
 
   var lastBeamConfig: BeamConfig = beamConfig
```

Comment:
This var should be either @volatile or atomic reference. But this trick decreases performance since during simulation agents refer to this lastBeamConfig. I think that if it worked before with a static field then we can leave it as it is for now.